scrapi/processing/elasticsearch.py

```
@@ -20,37 +22,65 @@
+        self.es = None
+        self.kwargs = {
+            'timeout': timeout or settings.ELASTIC_TIMEOUT,
+            'retry_on_timeout': True
+        }
+        self.kwargs.update(kwargs)
+
+    def setup(self):
+        '''Sets up the database connection. Returns True if the database connection
+            is successful, False otherwise
+        '''
+        try:
+            # If we cant connect to elastic search dont define this class
+            self.es = Elasticsearch(self.uri, **self.kwargs)
+
+            self.es.cluster.health(wait_for_status='yellow')
+            self.es.indices.create(index=self.index, body={}, ignore=400)
+            self.es.indices.create(index='share_v1', ignore=400)
+            return True
+        except ConnectionError:  # pragma: no cover
+            logger.error('Could not connect to Elasticsearch, expect errors.')
+            return False
 
-    es.cluster.health(wait_for_status='yellow')
-    es.indices.create(index=settings.ELASTIC_INDEX, body={}, ignore=400)
-    es.indices.create(index='share_v1', ignore=400)
-except ConnectionError:  # pragma: no cover
-    logger.error('Could not connect to Elasticsearch, expect errors.')
-    if 'elasticsearch' in settings.NORMALIZED_PROCESSING or 'elasticsearch' in settings.RAW_PROCESSING:
-        raise
+    def tear_down(self):
+        '''since it's just http, doesn't do much
+        '''
+        pass
+
+    def clear(self, force=False):
+        assert force, 'Force must be called to clear the database'
+        assert self.index != settings.ELASTIC_INDEX, 'Cannot erase the production database'
+        self.es.indices.delete(index=self.index, ignore=[400, 404])
+
+    def celery_setup(self, *args, **kwargs):
+        pass
 
 
 class ElasticsearchProcessor(BaseProcessor):
     NAME = 'elasticsearch'
 
-    def process_normalized(self, raw_doc, normalized, index=settings.ELASTIC_INDEX):
+    manager = DatabaseManager()
+
+    def documents(self, *sources):
+        pass
```

might want to just `raise NotImplementedError`. Fail fast, fail hard